### PR TITLE
Update adsd3500 reset

### PR DESCRIPTION
**Pull request by SeptimiuVana**

### CODE REVIEW

File: sdk/src/connections/target/adsd3500_sensor.cpp [hunk 1]
```diff
@@ -303,6 +297,69 @@ aditof::Status Adsd3500Sensor::open() {
...
         }
+
+        //Check chip status and reset if there are any errors.
+        if (m_firstRun) {
+            aditof::Status chipStatus = aditof::Status::GENERIC_ERROR;
+            uint16_t chipID;
+            uint8_t dealiasCheck[32] = {0};
+            dealiasCheck[0] = 1;
+
+            while (chipStatus != aditof::Status::OK) {
```
Comment (dNechita) [inline sdk/src/connections/target/adsd3500_sensor.cpp:309]: I wouldn't loop forever if reading chip id continues to fail. It can try 10 times for example. Sometimes there might be other issues with the hardware, preventing the chip id to be read. And in that case we don't want any of our programs to hang.

File: sdk/src/connections/target/adsd3500_sensor.cpp [hunk 2]
```diff
@@ -115,10 +115,11 @@ struct Adsd3500Sensor::ImplData {
 // Should not have duplicated code if possible.
 static int xioctl(int fh, unsigned int request, void *arg) {
     int r;
+    int tries = 3;
```
Comment (dNechita) [inline sdk/src/connections/target/adsd3500_sensor.cpp:118]: Why the need to change?

File: sdk/src/connections/target/adsd3500_sensor.cpp [hunk 3]
```diff
@@ -297,12 +292,40 @@ aditof::Status Adsd3500Sensor::open() {
...
+                chipStatus = adsd3500_read_payload_cmd(0x02, dealiasCheck, 32);
+                if (chipStatus != Status::OK) {
+                    LOG(ERROR) << "Failed to read dealias parameters for "
+                                  "adsd3500. Reseting chip...";
+                    adsd3500_reset();
+                }
+            }
+            m_firstRun = false;
+
+            LOG(INFO) << "Chip programmed successfully!";
```
Comment (dNechita) [inline sdk/src/connections/target/adsd3500_sensor.cpp:327]: I would rephrase this since it sounds like a programming just happened. Maybe say something like:
"ADSD3500 is ready to communicate with"

File: sdk/src/connections/target/adsd3500_sensor.cpp (hunk 1 above)
Comment (dNechita) [reply sdk/src/connections/target/adsd3500_sensor.cpp:309]: And if reading chip id & dealias parameters fails after N attempts, then you could log (as error) that "Cannot read chip ID " or "Cannot read dealias parameters" depending on the situation. Then you could say: "Latest ADSD3500 programming might not have been successful"

File: sdk/src/connections/target/adsd3500_sensor.cpp (hunk 2 above)
Comment (SeptimiuVana) [reply sdk/src/connections/target/adsd3500_sensor.cpp:118]: ctrl_app uses the same retry mechanism, it seems more stable.

File: sdk/src/connections/target/adsd3500_sensor.cpp (hunk 1 above)
Comment (SeptimiuVana) [reply sdk/src/connections/target/adsd3500_sensor.cpp]: Updated.

**Review by dNechita — APPROVED** [review]
